inc/zoo/swar/SWAR.h

```diff
@@ -244,14 +245,20 @@ struct BooleanSWAR: SWAR<NBits, T> {
     /// BooleanSWAR treats the MSB of each lane as the boolean associated with that lane.
     /// A logical NOT in this circumstance _only_ flips the MSB of each lane.  This operation is
     /// not ones or twos complement.
-    constexpr auto operator not() const noexcept {
-        return BooleanSWAR(MaskLaneMSB ^ *this);
+    constexpr auto operator ~() const noexcept {
+        return BooleanSWAR(Base{Base::MostSignificantBit} ^ *this);
     }
 
+    #define BOOLEANSWAR_BINARY_LOGIC_OPERATOR X(^) X(&) X(|)
+    #define X(op) \
+        constexpr BooleanSWAR operator op(BooleanSWAR other) const noexcept { return this->Base::operator op(other); }
+    BOOLEANSWAR_BINARY_LOGIC_OPERATOR
+    #undef X
```

Comment:
use the canonical name `BOOLEANSWAR_BINARY_LOGIC_OPERATOR_X_LIST`

Reply:
are these notes to self?